src/Processors/Formats/Impl/JSONColumnsBaseBlockOutputFormat.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <Core/Block.h>
+#include <Processors/Formats/IOutputFormat.h>
+#include <Formats/FormatSettings.h>
+
+
+namespace DB
+{
+
+class WriteBuffer;
+
+/// Base class for Columnar JSON output formats.
+/// It buffers all data and outputs it as a single block in writeSuffix() method.
+class JSONColumnsBaseBlockOutputFormat : public IOutputFormat
```

Comment:
(minor: totally not sure how the convention is in CH but many people would like "Base" at the end - ```JSONColumnsBlockOutputFormatBase```. Same for ```JSONColumnsBaseReader``` vs ```JSONColumnsReaderBase```)